sky/packages/sky/lib/src/widgets/basic.dart

```diff
@@ -243,6 +243,38 @@ class Center extends Align {
     : super(key: key, shrinkWrap: shrinkWrap, child: child);
 }
 
+class CustomShiftedBox extends OneChildRenderObjectWidget {
```

Comment:
I wish this were called CustomLayout, like CustomPaint. Then I would have expected callbacks for computing the child's size, and for its position. We have getSize and getChildPosition. The names, at least on the face of it, seem inconsistent. Maybe just getSize and getPosition?  

I'm not sure why I need to provide a getInnerConstraints callback. Why isn't getSize() enough?

We use the qualifiers "inner" and "intrinsic" here and elsewhere. I'm not sure exactly what they mean. I would assume that and intrinsic quality is one that doesn't depend on other widgets. So "inner"? We seem to have entered the realm of the spiritual :-).